### PR TITLE
Add movable camera example and entry

### PR DESCRIPTION
**Pull request by elfeiin**
Has pitch and zoom constraints. God was the pitch hard to get right. This one relies on a parent transform to make some calculations easier.

### CODE REVIEW

File: src/code/examples/movable-camera.rs [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+      let mut rotation_move = Vec2::ZERO;
+      let mut scroll = 0.0;
+
+      if buttons.pressed(MouseButton::Right) {
+         for ev in motion.iter() {
+            rotation_move += ev.delta;
+         }
+      }
+
+      for ev in scroll_evr.iter() {
```
Comment (alice-i-cecile) [inline src/code/examples/movable-camera.rs:230]:
```suggestion
      for event in scroll_events.iter() {
```

Characters are cheap ;)

**Review by alice-i-cecile — APPROVED** [review]
This is a really solid and useful snippet! I suspect that the rather-involved geometric calculations will be helpful examples for other use cases too.

In real code, I would probably strongly consider splitting movable_camera into several systems, and hiding it begind a plugin. I'm unsure how I feel about it here.

File: src/code/examples/movable-camera.rs [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+use std::ops::{Div, Mul, Neg};
+
+use bevy::{
+   input::mouse::{MouseMotion, MouseWheel},
+   prelude::*,
+};
+
+// ANCHOR: example
+/// Trait for customizing camera behavior.
+pub trait MovableCameraParams {
```
Comment (alice-i-cecile) [inline src/code/examples/movable-camera.rs:12]: This should just be stored as a resource struct: no reason to overcomplicate things, and that makes it configurable.

Comment (alice-i-cecile) [reply src/code/examples/movable-camera.rs:12]: Or actually, as a bundle of components on the camera entity.

This really wants a real input mapping solution, but that's out-of-scope.

Comment (elfein727) [reply src/code/examples/movable-camera.rs:12]: oop, you're right, i just didn't know how to do that. i can fix it when I'm awake.

**Pull request closed without merging**